src/domain/usecases/GetCurrentUserUseCase.ts

```diff
@@ -0,0 +1,10 @@
+import { User } from "../entities/User";
+import { UsersRepository } from "../repositories/UsersRepository";
+
+export class GetCurrentUserUseCase {
```

Comment:
Note: This use case is called and then more logic is done. It's better that the use case (with a new and descriptive name) performs all the logic. No need to change it.